exciting_environments/core_env.py

```diff
@@ -114,73 +157,192 @@ def create_in_axes_env_properties(self):
-            states, action, self.env_properties
+        obs, reward, terminated, truncated, state = jax.vmap(self.step, in_axes=(0, 0, self.in_axes_env_properties))(
+            state, action, self.env_properties
+        )
+        return obs, reward, terminated, truncated, state
+
+    @partial(jax.jit, static_argnums=[0, 4, 5])
+    def sim_ahead(self, init_state, actions, env_properties, obs_stepsize, action_stepsize):
+        """Computes multiple JAX-JIT compiled simulation steps for one batch.
+
+        The length of the set of inputs together with the action_stepsize determine the
+        overall length of the simulation -> overall_time = actions.shape[0] * action_stepsize
+        The actions are interpolated with zero order hold inbetween their values.
+
+        Args:
+            init_state: The initial state of the simulation
+            actions: A set of actions to be applied to the environment, the value changes every
+            action_stepsize (shape=(n_action_steps, action_dim))
+            env_properties: The constant properties of the simulation
+            obs_stepsize: The sampling time for the observations
+            action_stepsize: The time between changes in the input/action
+        """
+
+        assert actions.ndim == 2, "The actions need to have two dimensions: (n_action_steps, action_dim)"
+        assert (
+            actions.shape[-1] == self.action_dim
+        ), f"The last dimension does not correspond to the action dim which is {self.action_dim}, but {actions.shape[-1]} is given"
+
+        init_physical_state_shape = jnp.array(tree_flatten(init_state.physical_state)[0]).T.shape
+        assert init_physical_state_shape == (self.physical_state_dim,), (
+            "The initial physical state needs to be of shape (env.physical_state_dim,) which is "
+            + f"{(self.physical_state_dim,)}, but {init_physical_state_shape} is given"
+        )
+
+        # denormalize actions
+        actions = actions * jnp.array(tree_flatten(self.env_properties.action_constraints)[0]).T
+
+        # compute states trajectory for given actions
+        states = self._ode_solver_simulate_ahead(
+            init_state, actions, env_properties.static_params, obs_stepsize, action_stepsize
+        )
+
+        # generate observations for all timesteps
+        observations = jax.vmap(
+            self.generate_observation, in_axes=(0, self.in_axes_env_properties.physical_constraints)
+        )(states, self.env_properties.physical_constraints)
+
+        # generate rewards - use obs[1:] because obs[0] is observation for the initial state
+        reward = jax.vmap(self.reward_func, in_axes=(0, 0, self.in_axes_env_properties.action_constraints))(
+            observations[1:],
+            jnp.expand_dims(jnp.repeat(actions, int(action_stepsize / obs_stepsize)), 1),
+            self.env_properties.action_constraints,
         )
```

Comment:
I see no reason to not also compute the reward for the initial observation. If someone does not want to use it, they could still discard it. Wdyt?

Reply:
The action is sometimes taken into account in the reward. We wouldn't have any action for the initial observation. Would be a little tricky then. I think we had also discussed that the reward should no longer be calculated at this point anyway and will be calculated later outside the env in the gym_wrapper. I don't know how we want to keep this for the push now.

Reply:
Yes I agree. For time being the rewards will not be touched. 
We will have to decide if we want to keep the option for a reward.

As we discussed the rewards should not be part of the internal simulation. Likely they will be an option for the wrapper level.

Reply:
Any user that wants to use the framework for RL will likely implement their own reward anyways (?)